File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.js
// Last updated: 7/30/2026, 10:07:15 AM
var findDuplicate = function(nums) {
    nums.sort((a,b)=>a-b);

    for(let i=1;i<nums.length;i++){
        if(nums[i]===nums[i-1]){
            return nums[i];
        }
    }
};

File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.cpp
// Last updated: 7/30/2026, 10:07:09 AM
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int, int> frequency;  
        for(int i = 0; i < nums.size(); i++) {
            if(frequency[nums[i]] == 0) {
                frequency[nums[i]] = 1;
            }
            else {
                return nums[i]; 
            }
        }

        return -1;
    }
};